frontend/packages/dev-console/src/components/dropdown/ApplicationDropdown.tsx

```diff
@@ -41,86 +46,73 @@ interface ApplicationDropdownProps {
-      kind: CronJobModel.kind,
-      namespace,
-      prop: 'cronjobs',
-      optional: true,
-    },
-    {
-      isList: true,
-      kind: JobModel.kind,
-      namespace,
-      prop: 'jobs',
-      optional: true,
-    },
-    {
-      isList: true,
-      kind: PodModel.kind,
-      namespace,
-      prop: 'pods',
-      optional: true,
-    },
-    ...getDynamicChannelResourceList(namespace),
-    ...getDynamicEventSourcesResourceList(namespace),
-    ...knativeEventingResourcesBroker(namespace),
-  ];
+const applicationSelector = ['metadata', 'labels', 'app.kubernetes.io/part-of'];
+
+const ApplicationDropdown: React.FC<ApplicationDropdownProps & StateProps> = ({
+  namespace,
+  kindsInFlight,
+  ...props
+}) => {
+  const { t } = useTranslation();
+  const [resources, setResources] = React.useState<FirehoseResult[]>([]);
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [loadError, setLoadError] = React.useState();
+  const modelFactories = useExtensions<TopologyDataModelFactory>(isTopologyDataModelFactory);
+
+  const watchedResources = React.useMemo(() => {
+    const watchedBaseResources = getBaseWatchedResources(namespace);
+    return modelFactories.reduce((acc, modelFactory) => {
+      const factoryResources: WatchK8sResources<any> = modelFactory.properties.resources
+        ? modelFactory.properties.resources(namespace)
+        : {};
+      return { ...acc, ...factoryResources };
+    }, watchedBaseResources);
+  }, [modelFactories, namespace]);
+
+  const watchResults = useK8sWatchResources(watchedResources);
+  React.useEffect(() => {
+    const resourcesLoaded =
+      !kindsInFlight &&
+      Object.keys(watchResults).length > 0 &&
+      Object.keys(watchResults).every(
+        (key) => watchResults[key].loaded || watchResults[key].loadError,
+      ) &&
+      !Object.keys(watchResults).every((key) => watchResults[key].loadError);
+    if (!resourcesLoaded) {
+      return;
+    }
+
+    const optionalResources = Object.keys(watchedResources).filter(
+      (key) => watchedResources[key].optional,
+    );
+
+    const loadErrorKey = Object.keys(watchResults).find(
+      (key) => watchResults[key].loadError && !optionalResources.includes(key),
+    );
+    setLoadError(loadErrorKey ? watchResults[loadErrorKey].loadError : null);
+    setLoaded(resourcesLoaded);
+    const loadedResources = Object.keys(watchResults).map((key) => ({
+      ...watchResults[key],
+    }));
+    setResources(loadedResources);
+  }, [kindsInFlight, watchResults, watchedResources]);
+
   return (
-    <Firehose resources={resources}>
-      <ResourceDropdown
-        {...props}
-        placeholder="Select an Application"
-        dataSelector={['metadata', 'labels', 'app.kubernetes.io/part-of']}
-      />
-    </Firehose>
+    <ResourceDropdown
+      {...props}
+      resources={resources}
+      loaded={loaded}
+      loadError={loadError}
+      placeholder={t('devconsole~Select application')}
```

Comment:
```suggestion
      placeholder={t('devconsole~Select an application')}
```
unsure if we need to take care of localization in this PR 'cause PR https://github.com/openshift/console/pull/6905 takes care of all the files under dev-console

Reply:
afaik, the ask was to make sure any PRs do the i18n work as well in order to help ensure things are not missed.